docs/scripts/helpers.js

```diff
@@ -27,23 +27,23 @@ function fixLineEndings(source, target) {
 /**
  * Converts styled or regular component d.ts file to unstyled d.ts
  *
- * @param {string} tsFile - the definition file of the styled or regular mui component
+ * @param {string} filename - the file of the styled or regular mui component
  */
-function getUnstyledDefinitionFilename(tsFile) {
-  if (tsFile.indexOf('Unstyled') > -1) {
-    return tsFile;
+function getUnstyledFilename(filename, definitionFile = false) {
+  if (filename.indexOf('Unstyled') > -1) {
+    return filename;
   }
   let unstyledFile = '';
 
-  const separator = tsFile.indexOf('/') > -1 ? '/' : '\\';
+  const separator = filename.indexOf('/') > -1 ? '/' : '\\';
 
-  if (tsFile.endsWith('.d.ts') && tsFile.indexOf('material-ui-unstyled') === -1) {
-    unstyledFile = tsFile;
+  if (filename.indexOf('material-ui-unstyled') === -1) {
+    unstyledFile = filename.replace('.d.ts', '').replace('.ts', '').replace('.js', '');
     unstyledFile = unstyledFile.replace(/Styled/g, '');
 
     const pathParts = unstyledFile.split(separator);
 
-    const componentName = pathParts[pathParts.length - 1].replace('.d.ts', '');
+    const componentName = pathParts[pathParts.length - 1].replace('.d.ts', '').replace('.js', '');
```

Comment:
Didn't we already remove the file extensions at Line 41 (https://github.com/mui-org/material-ui/pull/23614/files#diff-adc4edb8ecfbfd9903441732f4e1a16f212f38edafad24107013bea4c0c57ce0R41)?

Reply:
Correct, I did that change afterward and forgot to update this. Good catch, let me update it.